sys/kern/kern_procctl.c

```diff
@@ -929,6 +929,67 @@ wxmap_status(struct thread *td, struct proc *p, void *data)
 	return (0);
 }
 
+#ifdef CHERI_CAPREVOKE
+static int
+cheri_revoke_ctl(struct thread *td, struct proc *p, void *data)
+{
+	int state;
+
+	PROC_LOCK_ASSERT(p, MA_OWNED);
+	state = *(int *)data;
+
+	switch (state) {
+	case PROC_CHERI_REVOKE_FORCE_ENABLE:
+		p->p_flag2 &= ~P2_CHERI_REVOKE_DISABLE;
+		p->p_flag2 |= P2_CHERI_REVOKE_ENABLE;
+		break;
+	case PROC_CHERI_REVOKE_FORCE_DISABLE:
+		p->p_flag2 |= P2_CHERI_REVOKE_DISABLE;
+		p->p_flag2 &= ~P2_CHERI_REVOKE_ENABLE;
+		break;
+	case PROC_CHERI_REVOKE_NOFORCE:
+		p->p_flag2 &=
+		    ~(P2_CHERI_REVOKE_ENABLE | P2_CHERI_REVOKE_DISABLE);
+		break;
+	default:
+		return (EINVAL);
+	}
+	return (0);
+}
+
+static int
+cheri_revoke_status(struct thread *td, struct proc *p, void *data)
+{
+	struct vmspace *vm;
+	int d;
+
+	switch (p->p_flag2 &
+	    (P2_CHERI_REVOKE_ENABLE | P2_CHERI_REVOKE_DISABLE)) {
+	case 0:
+		d = PROC_CHERI_REVOKE_NOFORCE;
+		break;
+	case P2_CHERI_REVOKE_ENABLE:
+		d = PROC_CHERI_REVOKE_FORCE_ENABLE;
+		break;
+	case P2_CHERI_REVOKE_DISABLE:
+		d = PROC_CHERI_REVOKE_FORCE_DISABLE;
+		break;
+	}
```

Comment:
Worth some kind of assertion both aren't set, whether before the switch or in default?